### ISSUE TITLE
Memory issues when setting up the OracleDatabase vagrant box

### ISSUE THREAD

**Issue by juli1185** [oracle/vagrant-projects#218]
Hello,

I set the memory to 2 GB (2048 MB) in the /vagrant-boxes/OracleDatabase/19.3.0/ora-response/dbca.rsp.tmpl and ran a 'vagrant up' command and noticed the following:

    oracle-19c-vagrant: [WARNING] [DBT-11209] Current available memory is less than the required available memory (2,048MB) for creating the database.
    oracle-19c-vagrant:    CAUSE: Following nodes do not have required available memory :
    oracle-19c-vagrant:  Node:oracle-19c-vagrant		Available memory:1.3842GB (1451420.0KB)
    oracle-19c-vagrant: 
    oracle-19c-vagrant: [WARNING] [DBT-11205] Specified shared pool size does not meet the recommended minimum size requirement. This might cause database creation to fail.
    oracle-19c-vagrant:    ACTION: Specify at least (1,998 MB) for shared pool size.

The database does come up but if you try to do a shutdown via sqlplus and a startup it does not allow you to start the database back up.

Please review and give some suggestions on how to fix this issue.

Thank you

Julius Pelejo
Sr. Ecommerce Architect

**Comment by PaulNeumann** [comment]
@juli1185 A quick way to address this is to increase the amount of memory specified in the `Vagrantfile` file. If you open the `Vagrantfile` file in a text editor, and change line 48 from

```ruby
    v.memory = 2048
```
to

```ruby
    v.memory = 4096
```
that will increase the VM's memory from 2 GB to 4 GB. You could increase it even further, if you need to (and your host has enough available RAM).

I hope this helps.

Regards,
Paul

**Comment by juli1185** [comment]
Hello Paul,

I increase the memory size in the VagrantFile from 2 GB to 4 GB but I still see the following:

    oracle-19c-vagrant: [WARNING] [DBT-11205] Specified shared pool size does not meet the recommended minimum size requirement. This might cause database creation to fail.
    oracle-19c-vagrant:    ACTION: Specify at least (1,998 MB) for shared pool size.

How do I resolve this issue?

Please advise

Thank you

Julius Pelejo
Sr. Ecommerce Architect

**Comment by PaulNeumann** [comment]
@juli1185 I suggest changing `totalMemory` back to 1536 in `dbca.rsp.tmpl`, then rebuilding the box (`vagrant destroy`, `vagrant up`). Using the original files from this repository (meaning that the VM has only 2 GB of RAM), I can shut down and start the database successfully using SQL*Plus:

```
    oracle-19c-vagrant: ORACLE PASSWORD FOR SYS, SYSTEM AND PDBADMIN: knwbq/82QJw=1
    oracle-19c-vagrant: INSTALLER: Installation complete, database ready to use!
PS C:\vagrant-boxes\OracleDatabase\19.3.0> vagrant ssh
getting Proxy Configuration from Host...

Welcome to Oracle Linux Server release 7.8 (GNU/Linux 4.14.35-1902.300.11.el7uek.x86_64)

The Oracle Linux End-User License Agreement can be viewed here:

  * /usr/share/eula/eula.en_US

For additional packages, updates, documentation and community help, see:

  * https://yum.oracle.com/

[vagrant@oracle-19c-vagrant ~]$ sudo su - oracle
Last login: Thu Apr 16 18:55:57 -06 2020
[oracle@oracle-19c-vagrant ~]$ sqlplus / as sysdba

SQL*Plus: Release 19.0.0.0.0 - Production on Thu Apr 16 19:02:42 2020
Version 19.3.0.0.0

Copyright (c) 1982, 2019, Oracle.  All rights reserved.


Connected to:
Oracle Database 19c Enterprise Edition Release 19.0.0.0.0 - Production
Version 19.3.0.0.0

SQL> shutdown
Database closed.
Database dismounted.
ORACLE instance shut down.
SQL> startup
ORACLE instance started.

Total System Global Area 1207955552 bytes
Fixed Size                  9134176 bytes
Variable Size             318767104 bytes
Database Buffers          872415232 bytes
Redo Buffers                7639040 bytes
Database mounted.
Database opened.
SQL> alter pluggable database all open;

Pluggable database altered.
SQL> 
```

Regards,
Paul

**Comment by juli1185** [comment]
Hello Paul, 

Can you show me what you have in your Vagrantfile and your dbca.rsp.tmpl file?   I adjusted my settings back to the 1536 and I got this:

    oracle-19c-vagrant: [WARNING] [DBT-11205] Specified shared pool size does not meet the recommended minimum size requirement. This might cause database creation to fail.
    oracle-19c-vagrant:    ACTION: Specify at least (1,998 MB) for shared pool size.
    oracle-19c-vagrant: Prepare for db operation
    oracle-19c-vagrant: 8% complete
    oracle-19c-vagrant: Copying database files
    oracle-19c-vagrant: 9% complete
    oracle-19c-vagrant: [WARNING] ORA-00821: Specified value of sga_target 768M is too small, needs to be at least 1216M
    oracle-19c-vagrant: ORA-01078: failure in processing system parameters
    oracle-19c-vagrant: [FATAL] ORA-01034: ORACLE not available
    oracle-19c-vagrant: 31% complete
    oracle-19c-vagrant: 100% complete
    oracle-19c-vagrant: [FATAL] ORA-01034: ORACLE not available
    oracle-19c-vagrant: 8% complete
    oracle-19c-vagrant: 0% complete
    oracle-19c-vagrant: Look at the log file "/opt/oracle/cfgtoollogs/dbca/ORCLCDB/ORCLCDB.log" for further details.
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.

Here is my Vagrantfile:

#
# LICENSE UPL 1.0
#
# Copyright (c) 1982-2018 Oracle and/or its affiliates. All rights reserved.
# 
# Since: July, 2018
# Author: gerald.venzl@oracle.com
# Description: Creates an Oracle database Vagrant virtual machine.
# 
# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
# 

# -*- mode: ruby -*-
# vi: set ft=ruby :

# Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
VAGRANTFILE_API_VERSION = "2"

# define hostname
NAME = "oracle-19c-vagrant"

unless Vagrant.has_plugin?("vagrant-proxyconf")
  puts 'Installing vagrant-proxyconf Plugin...'
  system('vagrant plugin install vagrant-proxyconf')
end

# get host time zone for setting VM time zone, if possible
# can override in env section below
offset_sec = Time.now.gmt_offset
if (offset_sec % (60 * 60)) == 0
  offset_hr = ((offset_sec / 60) / 60)
  timezone_suffix = offset_hr >= 0 ? "-#{offset_hr.to_s}" : "+#{(-offset_hr).to_s}"
  SYSTEM_TIMEZONE = 'Etc/GMT' + timezone_suffix
else
  # if host time zone isn't an integer hour offset, fall back to UTC
  SYSTEM_TIMEZONE = 'UTC'
end

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.box = "ol7-latest"
  config.vm.box_url = "https://yum.oracle.com/boxes/oraclelinux/latest/ol7-latest.box"
  config.vm.define NAME
  
  config.vm.box_check_update = false
  
  # change memory size
  config.vm.provider "virtualbox" do |v|
    v.memory = 2048
    v.name = NAME
  end

  # add proxy configuration from host env - optional
  if Vagrant.has_plugin?("vagrant-proxyconf")
    puts "getting Proxy Configuration from Host..."
    if ENV["http_proxy"]
      puts "http_proxy: " + ENV["http_proxy"]
      config.proxy.http     = ENV["http_proxy"]
    end
    if ENV["https_proxy"]
      puts "https_proxy: " + ENV["https_proxy"]
      config.proxy.https    = ENV["https_proxy"]
    end
    if ENV["no_proxy"]
      config.proxy.no_proxy = ENV["no_proxy"]
    end
  end

  # VM hostname
  config.vm.hostname = NAME

  # Oracle port forwarding
  config.vm.network "forwarded_port", guest: 1521, host: 1521
  config.vm.network "forwarded_port", guest: 5500, host: 5500

  # Provision everything on the first run
  config.vm.provision "shell", path: "scripts/install.sh", env:
    {
       "ORACLE_BASE"         => "/opt/oracle",
       "ORACLE_HOME"         => "/opt/oracle/product/19c/dbhome_1",
       "ORACLE_SID"          => "ORCLCDB",
       "ORACLE_PDB"          => "ORCLPDB1",
       "ORACLE_CHARACTERSET" => "AL32UTF8",
       "ORACLE_EDITION"      => "EE",
       "SYSTEM_TIMEZONE"     => SYSTEM_TIMEZONE
    }

end

And here is my dbca.rsp.tmpl file:

responseFileVersion=/oracle/assistants/rspfmt_dbca_response_schema_v19.0.0
gdbName=###ORACLE_SID###
sid=###ORACLE_SID###
databaseConfigType=SI
createAsContainerDatabase=true
numberOfPDBs=1
pdbName=###ORACLE_PDB###
pdbAdminPassword=###ORACLE_PWD###
templateName=General_Purpose.dbc
sysPassword=###ORACLE_PWD###
systemPassword=###ORACLE_PWD###
emConfiguration=DBEXPRESS
emExpressPort=5500
dbsnmpPassword=###ORACLE_PWD###
storageType=FS
characterSet=###ORACLE_CHARACTERSET###
nationalCharacterSet=AL16UTF16
automaticMemoryManagement=FALSE
totalMemory=1536
# Some init.ora parameters - disable auditing to save space, enable FS optimizations
initParams=audit_trail=none,audit_sys_operations=false,filesystemio_options=setall,commit_logging=batch,commit_wait=nowait,processes=5000

Thank you 

Julius Pelejo
Sr. Ecommerce Architect

**Comment by PaulNeumann** [comment]
@juli1185 I used the original files cloned from this repository, so you can view the `Vagrantfile` [here](https://github.com/oracle/vagrant-boxes/blob/master/OracleDatabase/19.3.0/Vagrantfile), and the `dbca.rsp.tmpl` file [here](https://github.com/oracle/vagrant-boxes/blob/master/OracleDatabase/19.3.0/ora-response/dbca.rsp.tmpl).

The files that you posted look OK. If you've modified any other files, please try undoing those changes. If that doesn't help, then I suggest starting over from the beginning (`vagrant destroy`, clone or download the repository again, then `vagrant up`). 

Regards,
Paul

**Comment by AmedeeBulle** [comment]
Is this still an issue?

Defaults have been updated (#268), all projects should run using default values.

**Comment by gvenzl** [comment]
This has been fixed with PR #268
